src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1375,8 +1399,9 @@ public boolean handleTokenExpiration(HttpServletRequest httpRequest, HttpServlet
         return true;
     }
 
-    private void redirectToLoginUrl(HttpServletRequest req, HttpServletResponse res) throws IOException {
-        if (req != null && (req.getSession(false) != null || Strings.isNullOrEmpty(req.getHeader("Authorization")))) {
+    private void redirectToLoginUrl(@NonNull HttpServletRequest req, @NonNull HttpServletResponse res)
```

Comment:
why the annotations? they contradict the method body. For example, if `res` is `NonNull` then 

```
if (res != null) {
            res.sendRedirect(Jenkins.get().getSecurityRealm().getLoginUrl());
        }
```
will happen always

Reply:
This might be based on a not so glorious comment by myself: https://github.com/jenkinsci/oic-auth-plugin/pull/491#discussion_r1895743841

I wanted to say: either tighten the contract by adding the @NonNull annotation (then there is no need for the null check) or leave the null check...

Sorry for not be more clear in my previous comment.